gapic/src/main/com/google/gapid/views/TracerDialog.java

```diff
@@ -582,6 +592,30 @@ private void updateOnDeviceChange(Settings settings, DeviceCaptureInfo dev) {
         updatePerfettoConfigLabel(settings);
       }
 
+      private void runValidationCheck(Models models, DeviceCaptureInfo dev, TraceTypeCapabilities config) {
+        if (dev != null && isPerfetto(config)) {
+          validationStatusLoader.startLoading();
+          validationStatusText.setText("Device is being Validated");
+          models.devices.validateDevice(dev, () -> {
+            setValidationStatus(dev.validationStatus);
+          });
+        }
+      }
+
+      private void setValidationStatus(boolean status) {
+        validationStatusLoader.updateStatus(status);
+        validationStatusText.setText("Validation " + (status ? "Passed" : "Failed" + VALIDATION_FAILED_LANDING_PAGE));
+        validationStatusLoader.stopLoading();
+        validationStatus.set(status);
+      }
+
+      private void resetValidationStatus() {
```

Comment:
Is there a way to rerun validation?

Reply:
Refreshing the device list will rerun it.